### PR TITLE
create new env variable for backend url

### LINKED ISSUE #101: Connect frontend to hosted backend
Two possible approaches:
- Replace all occurrences of `localhost:3000` with `process.env.BACKEND_URL` or similar
- Add `BACKEND_URL=localhost:3000` to the frontend .env file with the backend URL from Slack
or
- Replace all occurrences of `localhost:3000` with `process.env.PROD ? process.env.BACKEND_URL : localhost:3000`
- Not sure if `.PROD` is the right key but there is a key that tells us whether we're in production or development

### PR DESCRIPTION
**Pull request by yi-fang-yuan**
Closes #101

### CODE REVIEW

File: frontend/src/constants/constants.ts [hunk 1]
```diff
@@ -14,11 +14,13 @@ export const MTL_MAX_LATITUDE = 45.752154;
 export const MAPBOX_GEOCODING_API =
   "https://api.mapbox.com/geocoding/v5/mapbox.places/";
 export const MAPBOX_STYLE_URL = "mapbox://styles/mumap/ckhb8l6je01ki1ao4tabh8e2q";
-export const GET_ALL_ARTISTS_API = "http://localhost:3000/artist";
-export const GET_ALL_BOROUGH_API = "http://localhost:3000/borough";
-export const CREATE_MURAL_API = "http://localhost:3000/mural";
+export const GET_ALL_ARTISTS_API = `${process.env.REACT_APP_BACKEND_URL}/artist`;
+export const GET_ALL_BOROUGH_API = `${process.env.REACT_APP_BACKEND_URL}/borough`;
+export const CREATE_MURAL_API = `${process.env.REACT_APP_BACKEND_URL}/mural`;
+export const GET_ALL_COLLECTION = `${process.env.REACT_APP_BACKENDURL}/collection`;
```
Comment (tedspare) [inline frontend/src/constants/constants.ts:20]: typo in `...BACKEND_URL`

File: frontend/src/constants/constants.ts [hunk 2]
```diff
@@ -14,11 +14,13 @@ export const MTL_MAX_LATITUDE = 45.752154;
...
 export enum FORM {
   MURAL = "Mural",
   COLLECTION = "Collection",
   TOUR = "Tour"
 };
+export const BACKENDURL = process.env.REACT_APP_BACKEND_URL;
```
Comment (tedspare) [inline frontend/src/constants/constants.ts:26]: we should be able to remove this if it's unused

Comment (yi-fang-yuan) [reply frontend/src/constants/constants.ts]: removed

File: frontend/src/constants/constants.ts (hunk 1 above)
Comment (yi-fang-yuan) [reply frontend/src/constants/constants.ts]: fixed!

**Review by zhenglinxiao — APPROVED** [review]
LGTM :)

**Comment by tedspare** [comment]
It looks like the API still serves over HTTP (rather than HTTPS) (not your fault) so would you rather
- extend this PR to update the backend to serve HTTPS
or
- merge this PR and make that a new issue?

image

**Pull request merged**